Add rosdep rule for range-v3 library

miguelprada opened the pull request:
Rosdep key for [range-v3](https://ericniebler.github.io/range-v3/) library, which: _is the basis of the range support in C++20_.

I believe the ubuntu, debian and gentoo packages are correct. Package listings in the respective repositories can be found in:

- https://packages.ubuntu.com/search?keywords=range-v3&searchon=names&suite=eoan&section=all
- https://packages.debian.org/buster/librange-v3-dev
- https://packages.gentoo.org/packages/dev-cpp/range-v3

I'm not so sure about fedora. I could not find any related result in [fedora packages](https://apps.fedoraproject.org/packages/s/range-v3), but there's some result [here](https://src.fedoraproject.org/rpms/range-v3). I don't know what that means. Any help here would be appreciated.

clalancette commented on rosdep/base.yaml:
```diff
@@ -4994,6 +4994,11 @@ r-base-dev:
   gentoo: [dev-lang/R]
   ubuntu: [r-base-dev]
+range-v3:
+  debian: [librange-v3-dev]
```
This package is only available for debian buster and later.  We still support debian stretch for Melodic, so this key will have to be more specific.  See https://github.com/ros/rosdistro/blob/1cf2a0266f6abaaf86293772cb3f3570a2b8bf82/rosdep/base.yaml#L229 for an example on how to do this.

miguelprada replied:
Marked as unavailable for stretch. The package is available in stretch-backports repository, but I guess that doesn't count, right?

clalancette replied:
Correct, we can only depend on things in the main, security, universe, and multiverse repositories.

clalancette commented on rosdep/base.yaml:
```diff
@@ -4994,6 +4994,11 @@ r-base-dev:
+  debian: [librange-v3-dev]
+  fedora: [range-v3]
+  gentoo: [dev-cpp/range-v3]
+  ubuntu: [librange-v3-dev]
```
Same comment for ubuntu; this package is only available on eoan and later, but we still support xenial and bionic for Kinetic and Melodic, respectively.  This key will need to be more specific.

miguelprada replied:
It's available for [bionic](https://packages.ubuntu.com/bionic/librange-v3-dev) as well.

miguelprada commented:
Thanks for the review, @clalancette. I think I addressed your comments about the debian/ubuntu distribution specificity.

Re. the fedora key: there's one there already, but I'm not sure that's correct. Should it be specified differnently?

clalancette commented on rosdep/base.yaml:
```diff
@@ -4994,6 +4994,18 @@ r-base-dev:
+range-v3:
+  debian:
+    '*': [librange-v3-dev]
+    jessie: null
```
We don't need to add jessie here, as it is out-of-support.

miguelprada replied:
I added it because it's one of the _recommended_ supported platforms for Kinetic and I guessed it would be nice to include. It's gone now.

clalancette replied:
Yeah, we're a bit inconsistent there.  We support these other Debian/Ubuntu versions when we first release the ROS distributions.  Once they go out of support, we stop making changes to rosdep keys for them, but we don't remove them from the REP.  Anyway, looks good to me now, thanks for iterating.

miguelprada replied:
Got it. Thanks for the explanation :+1:

clalancette commented on rosdep/base.yaml:
```diff
@@ -4994,6 +4994,18 @@ r-base-dev:
+  ubuntu:
+    '*': [librange-v3-dev]
+    artful: null
+    wily: null
```
Same for artful and wily; since they are out-of-support, we don't need to add exceptions for them.

miguelprada replied:
Same here. Added those because they were in the list of supported platforms for Kinetic (wily) and Melodic (artful). Also removed these.
